gameoflife/src/test/kotlin/com/anaplan/engineering/kazuki/gameoflife/TestConway.kt

```diff
@@ -0,0 +1,303 @@
+        assertEquals(8, neighbourCount(around(mk_Point(0, 0)), mk_Point(0, 0)))
+    }
+
+    @Test
+    fun newCells() {
+        assertEquals(mk_Population(mk_Point(0, 0)), newCells(block - mk_Point(0, 0)))
+        assertEquals(mk_Population(), newCells(block))
+        assertEquals(mk_Population(mk_Point(0, 1), mk_Point(-1, -2), mk_Point(1, -1), mk_Point(-2, 0)), newCells(toad))
+    }
+
+    @Test
+    fun deadCells() {
+        assertEquals(mk_Population(mk_Point(1, 0), mk_Point(-1, 0)), deadCells(blinker))
+        assertEquals(mk_Population(), deadCells(block))
+        assertEquals(mk_Population(), deadCells(mk_Population()))
+        assertEquals(
+            mk_Population(mk_Point(0, 0), mk_Point(2, 0), mk_Point(-2, 0)),
+            deadCells(mk_Population(mk_Point(0, 0), mk_Point(2, 0), mk_Point(-2, 0)))
+        )
+    }
+
+    @Test
+    fun generation() {
+        assertEquals(block, generation(block))
+        assertEquals(mk_Population(mk_Point(0, 1), mk_Point(0, 0), mk_Point(0, -1)), generation(blinker))
+        assertEquals(mk_Population(), generation(mk_Population()))
+    }
+
+    @Test
+    fun generations() {
+        assertEquals(
+            mk_Population(), generations(
+                4, mk_Population(
+                    mk_Point(-3, -3), mk_Point(-2, -2), mk_Point(-1, -1),
+                    mk_Point(0, 0), mk_Point(1, 1), mk_Point(2, 2), mk_Point(3, 3)
+                )
+            )
+        )
+        assertEquals(block, generations(1, block))
+        assertEquals(block, generations(500, block))
+        assertEquals(toad, generations(4, toad))
+    }
+
+    @Test
+    fun offset() {
+        assertEquals(
+            mk_Population(mk_Point(0, 0), mk_Point(1, 0), mk_Point(0, -1), mk_Point(1, -1)),
+            offset(block, 1, 0)
+        )
+        assertEquals(mk_Population(), offset(mk_Population(), 3, 2))
+        assertEquals(
+            mk_Population(mk_Point(9, 10), mk_Point(10, 10), mk_Point(9, 9), mk_Point(10, 9)),
+            offset(block, 10, 10)
+        )
+    }
+
+    @Test
+    fun isOffset() {
+        assertEquals(
+            true, isOffset(
+                blinker, mk_Population(mk_Point(0, 1), mk_Point(1, 1), mk_Point(2, 1)), 2
+            )
+        )
+        assertEquals(true, isOffset(mk_Population(), mk_Population(), 1))
+        assertEquals(false, isOffset(block, block, 5))
+        assertEquals(false, isOffset(blinker, mk_Population(), 2))
+        assertEquals(
+            false, isOffset(
+                block, mk_Population(
+                    mk_Point(2, 0), mk_Point(3, 0),
+                    mk_Point(2, -1), mk_Point(3, -1)
+                ), 2
+            )
+        )
+        assertEquals(
+            true, isOffset(
+                block, mk_Population(
+                    mk_Point(2, 0), mk_Point(3, 0),
+                    mk_Point(2, -1), mk_Point(3, -1)
+                ), 3
+            )
+        )
+    }
+
+    @Test
+    fun periodN() {
```

Comment:
⛏️ These are just test functions, so they didn't need testing themselves. Anyway, it doesn't hurt to leave these in now that they've been written.